modules/reactor/doc/content/source/meshgenerators/HexagonMeshTrimmer.md

```diff
@@ -0,0 +1,44 @@
+
+## Trimmability
+
+In general, `HexagonMeshTrimmer` trims meshes generated by `PatternedHexMeshGenerator`. An assembly mesh consisting of patterned pin meshes has both peripheral and center trimmability; whereas a core mesh consisting of patterned assembly meshes only has center trimmability.  Two mesh metadata entries (`peripheral_trimmability` and `center_trimmability`) are created by `PatternedHexMeshGenerator` to tell `HexagonMeshTrimmer` which trimming options are valid. In the absence of these two meta data, `HexagonMeshTrimmer` will throw an incompatible error message.
+
+In addition, `PeripheralRingMeshGenerator`, which applies quadrilateral peripheral cylinder mesh around an assembly or core, retain these two mesh meta data from the input mesh so that valid meshes generated by `PeripheralRingMeshGenerator` may also be trimmed. Similarly, assembly meshes modified by `PatternedHexPeripheralModifier` also have these meta data retained and are thus able to be trimmed by this generator.
+
+## Degenerate Quadrilateral Elements
+
+When trimming a mesh, some elements may be located across the trimming line and thus need to be processed to ensure a smooth trimming boundary. Generic tools such as [`PlaneDeletionGenerator`](/PlaneDeletionGenerator.md) is capable of trimming meshes but leaves a zigzag trimming boundary in the presence of across-trimming-line elements.
+
+To avoid this zig-zag boundary, `HexagonMeshTrimmer` adopts a post-trimming processing algorithm to smooth the trimming boundary. The algorithm moves the nodes of the across-trimming-line elements in the normal direction of the trimming line onto the trimming line. During this procedure, some elements may become zero volume and will be removed. More importantly, after node moving, some quadrilateral elements may have three co-linear vertices on the trimming line, which make the element degenerate. To fix this issue, these degenerate quadrilateral elements are converted into triangular elements. As triangular elements and quadrilateral elements cannot share a single subdomain id/name, new subdomains are created for any affected quadrilateral element subdomains. The subdomain ids of the new subdomains are decided by shifting the original subdomain ids by [!param](/Mesh/HexagonMeshTrimmer/tri_elem_subdomain_shift) (default shift value is the maximum subdomain id of the mesh), while the subdomain names of the new subdomains are created by appending [!param](/Mesh/HexagonMeshTrimmer/tri_elem_subdomain_name_suffix) after the original subdomain names.
+
```

Comment:
Add a small paragraph to explain the optional external boundary treatment, users could miss this